agent-langgraph/README.md

```diff
@@ -0,0 +1,226 @@
+     # DATABRICKS_HOST="https://host.databricks.com"
+     # DATABRICKS_TOKEN="dapi_token"
+     ```
+
+   See the [Databricks SDK authentication docs](https://docs.databricks.com/aws/en/dev-tools/sdk-python#authenticate-the-databricks-sdk-for-python-with-your-databricks-account-or-workspace) for more info.
+
+3. **Create and link an MLflow experiment to your app**
+
+   To enable MLflow tracing and version tracking, create an MLflow experiment in Databricks. This is automatically done by the `./scripts/quickstart.sh` script.
+
+   - **Manual setup**
+     Create the MLflow experiment manually via the CLI.
+
+     ```bash
+     DATABRICKS_USERNAME=$(databricks current-user me | jq -r .userName)
+     databricks experiments create-experiment /Users/$DATABRICKS_USERNAME/agents-on-apps
+     ```
+
+     Make a copy of `.env.example` to `.env.local` and update the `MLFLOW_EXPERIMENT_ID` in your `.env.local` file with the experiment ID you created. The `.env.local` file will be automatically loaded when starting the server.
+
+     ```bash
+     cp .env.example .env.local
+     # Edit .env.local and fill in your experiment ID
+     ```
+
+   Refer to the [MLflow experiments documentation](https://docs.databricks.com/aws/en/mlflow/experiments#create-experiment-from-the-workspace) for more info.
+
+4. **Testing out your local agent**
+
+   Start up the agent server and chat UI locally:
+
+   ```bash
+   ./scripts/start-app.sh
+
+   # You can also manually start the agent server with additional options:
+   uv run start-server --reload # hot-reload the server on code changes
+
+   # Other options for the start-server script:
+   uv run start-server --port 8001 # change the port the server listens on
+   uv run start-server --workers 4 # run the server with multiple workers
+   ```
+
+   Now you can either query your agent via the built in UI (served by default at http://localhost:8000) or via REST API request:
+
+   - Example streaming request:
+     ```bash
+     curl -X POST http://localhost:8000/invocations \
+     -H "Content-Type: application/json" \
+     -d '{ "input": [{ "role": "user", "content": "hi" }], "stream": true }'
+     ```
+   - Example non-streaming request:
+     ```bash
+     curl -X POST http://localhost:8000/invocations  \
+     -H "Content-Type: application/json" \
+     -d '{ "input": [{ "role": "user", "content": "hi" }] }'
+     ```
+
+5. **Modifying your agent**
+
+   You can check out the [LangGraph documentation](https://docs.langchain.com/oss/python/langgraph/quickstart) for more information on how to edit your own agent.
+
+   The following files are required to host your own agent with the MLflow `AgentServer`:
+
+   - `agent.py`: This file contains your agent logic. It currently contains a Responses API agent. Please modify this file to create your custom agent.
+   - `start_server.py`: This file initializes and runs the MLflow `AgentServer` with agent_type="ResponsesAgent".
+
+   Common changes to make:
```

Comment:
It may be good to make this more of an FAQ and put example questions that users might ask -> include these bullets as answers. For example, it's not immediately obvious to me why I would want to "add  files or folders to my agent" (but the reason to do that might be to modularize or add more logic to the agent etc)